MockServer - Refresh metadata on setting entity data

olirogers opened the pull request:
When we edit the entity set data, for example in response to a mocked function import, the metadata gets out of sync. especially if you add new entities.

The _enhanceWithMetadata function is added to setEntitySetData in order to refresh the __metadata properties of the changed entities.

Any comments or criticisms welcome.

Oli

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SAP/openui5?pullRequest=922) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

akudev commented:
Hi Oliver,
the owner of the code likes your code change, but would prefer to have a unit test covering this case. Would it be possible to add one?

Thanks
Andreas

olirogers commented:
Hi Andreas,

Leave it with me and I will have a look at how the current tests are structured and see what I can add.

Thanks for the response.

Oli

akudev commented:
Hi Oli, will you do something more here or do you prefer to abandon this change?

olirogers commented:
hi @akudev ,
Apologies for the delay! I have added a unit test for this into the MockServer.qunit.html and rebased on master.
Let me know if this is OK to merge or if there is anything else you need for this.
Oli

akudev commented:
Contacted the new mock server owners.

Shtilianova commented:
Hello @olirogers ,
Thank you for your pull request. I've created an internal record 1780292068. You can check its status on GitHub.
Regards,

olirogers commented:
Hi @Shtilianova, thanks for looking at this.

You've added the author action label but I am not sure what you need me to do? Let me know, thanks!

Oli

stephania87 commented:
@olirogers Apologies for the caused confusion. Labels mean an issue is followed by a member of SAP organization. If additional information is needed, we add comments.
Also when there are updates, they are posted in the corresponding GitHub thread.
We are all set for now :)

akudev commented:
Hi Oli,

it's been a while since this started, but yesterday your commit was merged by the colleagues owning the MockServer code: https://github.com/SAP/openui5/commit/95ea2b00484abb03dc5c9ad149f3ed24635cff1a
(We can't merge pull requests directly within GitHub as all changes must go through our internal code review system. This is why I just close this pull request now. Nevertheless it was your original commit that was merged.) 
It will be part of OpenUI5 1.54.

Thanks for your contribution!

Regards
Andreas

i059687 commented:
The contribution has been reverted since it caused the regression.
The method setEntitySetData   always override property _deferred, which sometimes was not required at all, and caused not correct behavior.
Regression description can be found here as well (https://github.com/SAP/openui5/issues/1851).


Regards,
Alla
